Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,320 @@
+/-
+Copyright (c) 2024 Colin Jones. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Colin Jones
+-/
+import Mathlib.NumberTheory.Divisors
+import Mathlib.Tactic.NormNum.Prime
+import Mathlib.Algebra.GeomSum
+import Mathlib.Tactic.FinCases
+
+/-!
+## FactorisationProperties.lean
+This file defines abundant, semi-perfect, deficient, and weird numbers and formalizes their
+  relations with prime and perfect numbers.
+
+## Main Definitions
+Let `n : ℕ`. All the following definitions are in the Nat namespace.
+* `Abundant` a natural number `n` is abundant if the sum of its proper divisors are greater than
+  itself and it is greater than zero
+* `Pseudoperfect` a natural number `n` is semi-perfect if a subset of its proper divisors equals
+  itself and it is greater than zero
+* `Deficient` a natural number `n` is deficient if the sum of its proper divisors is less than
+  itself and it is greater than zero
+* `WeirdNumber` a natural number is weird if it is both abundant and not semi-perfect
+
+## Main Results
+* `divisors_eq_proper_union_self` the `Finset` that is `Nat.divisors` is equal to the union of
+  `Nat.properDivisors` and the number itself
```

Comment:
```suggestion
* `Nat.divisors_eq_proper_union_self`: `Nat.divisors n` is the union of `Nat.properDivisors` and `{n}`

Reply:
One example: notable, I'd name the lemma in its namespace, and put a colon after its name.
In this case, I think the description could be more concise without losing information.

Reply:
Okay I see how to implement these suggestions, will update.